.pre-commit-config.yaml

```diff
@@ -9,34 +9,46 @@ ci:
 
 repos:
   - repo: https://github.com/asottile/pyupgrade
-    rev: v3.15.0
+    rev: v2.31.0
     hooks:
       - id: pyupgrade
-        args: ["--py37-plus"]
+        args: ["--py39-plus"]
 
   - repo: https://github.com/adamchainz/django-upgrade
-    rev: '1.15.0'
+    rev: '1.4.0'
     hooks:
       - id: django-upgrade
-        args: [--target-version, "2.2"]
-
-  - repo: https://github.com/PyCQA/flake8
-    rev: 6.1.0
-    hooks:
-      - id: flake8
+        args: [--target-version, "3.2"]
 
   - repo: https://github.com/asottile/yesqa
-    rev: v1.5.0
+    rev: v1.3.0
     hooks:
       - id: yesqa
 
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.5.0
+    rev: v4.1.0
```

Comment:
```suggestion
    rev: v4.5.0
```

Reply:
I'm making these modify and upgrading also codespell and ruff-pre-commit